ironfish/src/wallet/account/encoder/generic.ts

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { AccountImport } from '../../walletdb/accountValue'
+import { Bech32Encoder } from './bech32'
+import { Bech32JsonEncoder } from './bech32json'
+import {
+  AccountDecodingOptions,
+  AccountEncoder,
+  AccountEncodingOptions,
+  Formats,
+} from './encoder'
+import { JsonEncoder } from './json'
+import { MnemonicEncoder } from './mnemonic'
+import { SpendingKeyEncoder } from './spendingKey'
+
+export class GenericEncoder implements AccountEncoder {
```

Comment:
This seems ok, but is this better than Oreoscans importAccount() function? Does this make sense to be a class instead of just 2 functions? https://github.com/oreoslabs/oreos/blob/master/src/wallet/account.ts#L40-L53

Reply:
Thats fine with me, we thought it was convenient that this matches the class architecture of the other encoders, but I am not married to the generic encoder being a class.